lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -0,0 +1,950 @@
+#include "TypeChecker.h"
+#include "swift/AST/Decl.h"
+#include "swift/AST/Expr.h"
+#include "swift/AST/Module.h"
+#include "swift/AST/ParameterList.h"
+#include "swift/AST/Pattern.h"
+#include "swift/AST/Stmt.h"
+#include "swift/AST/Types.h"
+#include "DerivedConformances.h"
+
+using namespace swift;
+using namespace DerivedConformance;
+
+// Returns whether the type represented by the given ClassDecl inherits from a
+// type which is Codable.
+static bool superclassIsCodable(TypeChecker &tc, ClassDecl *type) {
+  assert(type && "Cannot get superclass of null type.");
+  if (!type->hasSuperclass())
+    return false;
+
+  ASTContext &C = tc.Context;
+  auto parentDC = type->getDeclContext();
+  auto superType = type->getSuperclass();
+  return (bool)tc.conformsToProtocol(superType,
+                                     C.getProtocol(KnownProtocolKind::Codable),
+                                     parentDC,
+                                     ConformanceCheckFlags::Used);
+}
+
+// Looks up a type declaration with the given name in the Foundation module.
+//
+// Asserts that the Foundation module is loaded, that the given name does not
+// refer to more than one type, and that the entity with the given name refers
+// to a type instead of a different type of declaration.
+static TypeDecl* lookupFoundationTypeDecl(ASTContext &C, Identifier name) {
```

Comment:
This violates Swift style. We use 'TypeDecl *'. This is a case that would be caught by clang-format.